include/pops/simulation.hpp

```diff
@@ -190,38 +190,57 @@ class Simulation
+     *
+     * @param infected Currently infected hosts
+     * @param total_hosts All hosts
+     * @param mortality_rate percent of infected hosts that die each time period
+     * @param mortality_time_lag time lag prior to mortality beginning
+     * @param died dead hosts during time step
+     * @param mortality_tracker_vector vector of matrices for tracking infected
+     * host infection over time.
+     * @param suitable_cells used to run model only where host are known to occur
+     */
     void mortality(
         IntegerRaster& infected,
+        IntegerRaster& total_hosts,
         double mortality_rate,
-        int current_year,
-        int first_mortality_year,
-        IntegerRaster& mortality,
+        int mortality_time_lag,
+        IntegerRaster& died,
         std::vector<IntegerRaster>& mortality_tracker_vector,
         const std::vector<std::vector<int>>& suitable_cells)
     {
-        if (current_year >= (first_mortality_year)) {
-            int mortality_current_year = 0;
-            int max_year_index = current_year - first_mortality_year;
 
-            for (auto indices : suitable_cells) {
-                int i = indices[0];
-                int j = indices[1];
-                for (int year_index = 0; year_index <= max_year_index; year_index++) {
-                    int mortality_in_year_index = 0;
-                    if (mortality_tracker_vector[year_index](i, j) > 0) {
-                        mortality_in_year_index =
-                            mortality_rate * mortality_tracker_vector[year_index](i, j);
-                        mortality_tracker_vector[year_index](i, j) -=
-                            mortality_in_year_index;
-                        mortality(i, j) += mortality_in_year_index;
-                        mortality_current_year += mortality_in_year_index;
-                        if (infected(i, j) > 0) {
-                            infected(i, j) -= mortality_in_year_index;
-                        }
+        int mortality_current_step = 0;
+        int max_index = mortality_tracker_vector.size() - mortality_time_lag - 1;
+
+        for (auto indices : suitable_cells) {
+            int i = indices[0];
+            int j = indices[1];
+            for (int index = 0; index <= max_index; index++) {
+                int mortality_in_index = 0;
+                if (mortality_tracker_vector[index](i, j) > 0) {
+                    if (index == 0) {
+                        mortality_in_index = mortality_tracker_vector[index](i, j);
+                    }
+                    else {
+                        mortality_in_index =
+                            mortality_rate * mortality_tracker_vector[index](i, j);
+                    }
+                    mortality_tracker_vector[index](i, j) -= mortality_in_index;
+                    died(i, j) += mortality_in_index;
+                    mortality_current_step += mortality_in_index;
```

Comment:
The value seems to be unused like with `mortality_current_year`. Probably left over from debugging or from the idea that there would be two "died" handled by this function (one for year and one total). I suggest to remove the variable.

Reply:
Isn't the variable still defined above, i.e., defined but unused. No big deal, we can get it next time we improve compile warnings status.

Reply:
Good catch, I removed it.